azurerm/resource_arm_signalr_service.go

```diff
@@ -0,0 +1,212 @@
+		return fmt.Errorf("Error waiting for the result of creating or updating SignalR %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	read, err := client.Get(ctx, resourceGroup, name)
+	if err != nil {
+		return err
+	}
+	if read.ID == nil {
+		return fmt.Errorf("SignalR %q (Resource Group %q) ID is empty", name, resourceGroup)
+	}
+	d.SetId(*read.ID)
+
+	return resourceArmSignalRServiceRead(d, meta)
+}
+
+func resourceArmSignalRServiceRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).signalRClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resourceGroup := id.ResourceGroup
+	name := id.Path["SignalR"]
+
+	resp, err := client.Get(ctx, resourceGroup, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[DEBUG] SignalR %q was not found in Resource Group %q - removing from state!", name, resourceGroup)
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error getting SignalR %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	d.Set("name", name)
+	d.Set("resource_group_name", resourceGroup)
+	if location := resp.Location; location != nil {
+		d.Set("location", azureRMNormalizeLocation(*location))
+	}
+	if err = d.Set("sku", flattenSignalRServiceSku(resp.Sku)); err != nil {
+		return fmt.Errorf("Error setting `sku`: %+v", err)
+	}
+	if properties := resp.Properties; properties != nil {
+		d.Set("hostname", properties.HostName)
+		d.Set("ip_address", properties.ExternalIP)
+		d.Set("public_port", properties.PublicPort)
+		d.Set("server_port", properties.ServerPort)
+	}
+	flattenAndSetTags(d, resp.Tags)
+
+	return nil
+}
+
+func resourceArmSignalRServiceDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).signalRClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resourceGroup := id.ResourceGroup
+	name := id.Path["SignalR"]
+
+	future, err := client.Delete(ctx, resourceGroup, name)
+	if err != nil {
+		if !response.WasNotFound(future.Response()) {
+			return fmt.Errorf("Error deleting SignalR %q (Resource Group %q): %+v", name, resourceGroup, err)
+		}
+		return nil
+	}
+	if err := future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		if !response.WasNotFound(future.Response()) {
+			return fmt.Errorf("Error waiting for the deletion of SignalR %q (Resource Group %q): %+v", name, resourceGroup, err)
+		}
+	}
+
+	return nil
+}
+
+func expandSignalRServiceSku(input []interface{}) *signalr.ResourceSku {
+	v := input[0].(map[string]interface{})
+	return &signalr.ResourceSku{
+		Name:     utils.String(v["name"].(string)),
+		Capacity: utils.Int32(int32(v["capacity"].(int))),
+	}
+}
+
+func flattenSignalRServiceSku(input *signalr.ResourceSku) []interface{} {
+	result := make(map[string]interface{})
+	if input != nil {
```

Comment:
if input's nil we're going to set an empty sku block in the state (so there'll be an object there but it'll be empty) - can we update this to:

```
if input == nil {
  return []interface{}{}
}

...
```

this way it'll show that the Sku block isn't being returned from the API from the fact that Terraform will show the block being removed, rather than the fields being blank